packages/experimental/Icon/src/Icon.tsx

```diff
@@ -0,0 +1,109 @@
+import * as React from 'react';
+import {IconProps, SvgIconProps, FontIconProps} from './Icon.types';
+import * as ReactNative from 'react-native';
+import { Text } from '@fluentui-react-native/text';
+import {SvgUri} from 'react-native-svg'
+import * as assetRegistry from 'react-native/Libraries/Image/AssetRegistry';
+import { stagedComponent, useTheme, mergeProps, getMemoCache } from '@fluentui-react-native/framework';
+
+const rasterImageStyleCache = getMemoCache<ReactNative.ImageStyle>();
+
+function renderRasterImage(iconProps: IconProps) {
+  const { width, height } = iconProps;
+  const style = rasterImageStyleCache({
+    width: width,
+    height:height
+  }, [width, height])[0];
+
+  return (
+    <ReactNative.Image
+      source = {iconProps.rasterImageSource.src}
+      style = {style}
+    />
+  )
+}
+
+const fontStyleMemoCache = getMemoCache();
+
+function renderFontIcon(iconProps: IconProps) {
+  const fontSource: FontIconProps = iconProps.fontSource;
+
+  const style = fontStyleMemoCache({
+      fontSrcFile: fontSource.fontSrcFile,
```

Comment:
You can change this to a function, then bring the if style.fontSrcFile stuff inside the function, that way you only update the fontFamily the first time you calculate the style.